File: frontend/src/types/recommendation.ts
export type RecommendationStatus = "ACTIVE" | "DISMISSED" | "COMPLETED";
export type RecommendationPriority = "HIGH" | "MEDIUM" | "LOW";
export type RecommendationType =
  | "RUN_WARM_UP"
  | "RUN_QUICK_UPVOTE"
  | "RUN_CASUAL_READER"
  | "RUN_DEEP_READER"
  | "SYNC_PROFILE"
  | "REFRESH_SESSION"
  | "NO_ACTION";

export interface AccountRecommendation {
  id: string;
  account_id: string;
  recommendation_type: RecommendationType;
  priority: RecommendationPriority;
  title: string;
  description: string;
  recommended_template_id: string | null;
  status: RecommendationStatus;
  reason: Record<string, unknown>;
  created_at: string;
  updated_at: string;
}

File: backend/app/schemas/recommendation.py
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, ConfigDict

from app.models.enums import RecommendationPriority, RecommendationStatus, RecommendationType


class AccountRecommendationRead(BaseModel):
    id: UUID
    account_id: UUID
    recommendation_type: RecommendationType
    priority: RecommendationPriority
    title: str
    description: str
    recommended_template_id: UUID | None
    status: RecommendationStatus
    reason: dict[str, Any]
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)
